File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string>ans;
        stack<int>st;
        int j = 0;
        for(int i = 1; i<=n; i++) {
            st.push(i);
            if(target[j] == st.top()) {
                j++;
                ans.push_back("Push");
            }
            else{
                ans.push_back("Push");
                ans.push_back("Pop");
            }
            if(j == target.size())  break;
        }
        return ans;
    }
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        Stack<Integer>st = new Stack<>();
        List<String>ans = new ArrayList<>();
        int j = 0;
        for(int i = 1; i<=n; i++) {
            st.push(i);
            if(target[j] == st.peek()) {
                ans.add("Push");
                j++;
            }
            else{
                ans.add("Push");
                ans.add("Pop"); 
            }
            if(j == target.length)  break;
        }
        return ans;
    }
}